x/oracle/keeper/history_test.go

```diff
@@ -0,0 +1,579 @@
+	phStoreJuno, err = oracleKeeper.getHistoryEntryAtOrBeforeTime(ctx, "Juno", phEntrysJuno[0].PriceUpdateTime)
+	require.NoError(t, err)
+	require.Equal(t, phEntrysJuno[0], phStoreJuno)
+	phStoreJuno, err = oracleKeeper.getHistoryEntryAtOrBeforeTime(ctx, "Juno", phEntrysJuno[0].PriceUpdateTime.Add(time.Minute))
+	require.NoError(t, err)
+	require.Equal(t, phEntrysJuno[0], phStoreJuno)
+	phStoreJuno, err = oracleKeeper.getHistoryEntryAtOrBeforeTime(ctx, "Juno", phEntrysJuno[1].PriceUpdateTime)
+	require.NoError(t, err)
+	require.Equal(t, phEntrysJuno[1], phStoreJuno)
+	phStoreJuno, err = oracleKeeper.getHistoryEntryAtOrBeforeTime(ctx, "Juno", phEntrysJuno[1].PriceUpdateTime.Add(time.Minute))
+	require.NoError(t, err)
+	require.Equal(t, phEntrysJuno[1], phStoreJuno)
+
+	phStoresdjuno, err := oracleKeeper.getHistoryEntryBetweenTime(
+		ctx,
+		"Juno",
+		phEntrysJuno[0].PriceUpdateTime.Add(-time.Minute),
+		phEntrysJuno[2].PriceUpdateTime.Add(time.Minute),
+	)
+	require.NoError(t, err)
+	require.Equal(t, phStoresdjuno, phEntrysJuno)
+
+	phStoresdjuno, err = oracleKeeper.getHistoryEntryBetweenTime(
+		ctx,
+		"Juno",
+		phEntrysJuno[0].PriceUpdateTime.Add(-time.Minute),
+		phEntrysJuno[1].PriceUpdateTime,
+	)
+	require.NoError(t, err)
+	require.Equal(t, 2, len(phStoresdjuno))
+	require.Equal(t, phStoresdjuno[0], phEntrysJuno[0])
+	require.Equal(t, phStoresdjuno[1], phEntrysJuno[1])
+}
+
+func TestStoreAndGetNullHistoricalData(t *testing.T) {
+	timeNow := time.Now().UTC()
+	ctx, keepers := CreateTestInput(t, false)
+	oracleKeeper := keepers.OracleKeeper
+
+	phEntrys := []types.PriceHistoryEntry{
+		{
+			Price:           sdk.OneDec(),
+			VotePeriodCount: 10,
+			PriceUpdateTime: timeNow,
+		},
+		{
+			Price:           sdk.OneDec(),
+			VotePeriodCount: 11,
+			PriceUpdateTime: timeNow.Add(time.Minute * 2),
+		},
+		{
+			Price:           sdk.OneDec(),
+			VotePeriodCount: 12,
+			PriceUpdateTime: timeNow.Add(time.Minute * 4),
+		},
+	}
+
+	for _, phEntry := range phEntrys {
+		oracleKeeper.storeHistoricalData(ctx, "Denom", phEntry)
+	}
+
+	// below queries should all throw error as "Juno" denom does not exist
+	phStore, err := oracleKeeper.getHistoryEntryAtOrBeforeTime(ctx, "Juno", phEntrys[0].PriceUpdateTime.Add(-time.Minute))
+	require.Error(t, err)
+	require.Equal(t, types.PriceHistoryEntry{}, phStore)
+	phStore, err = oracleKeeper.getHistoryEntryAtOrBeforeTime(ctx, "Juno", phEntrys[0].PriceUpdateTime)
+	require.Error(t, err)
+	require.Equal(t, types.PriceHistoryEntry{}, phStore)
+	phStore, err = oracleKeeper.getHistoryEntryAtOrBeforeTime(ctx, "Juno", phEntrys[1].PriceUpdateTime)
+	require.Error(t, err)
+	require.Equal(t, types.PriceHistoryEntry{}, phStore)
+
+	phStores, err := oracleKeeper.getHistoryEntryBetweenTime(
+		ctx,
+		"Juno",
+		phEntrys[0].PriceUpdateTime.Add(-time.Minute),
+		phEntrys[2].PriceUpdateTime.Add(time.Minute),
+	)
+	require.NoError(t, err) // To discuss: should this throw error?
```

Comment:
Doesn't pass test due to `no values in range`. not sure why this is not coming up on the CI test

Reply:
the test is an error check?

Reply:
Yes, fails with "go test ./..." right now locally for me